x-pack/plugins/stack_connectors/server/connector_types/gen_ai/gen_ai.ts

```diff
@@ -71,21 +80,34 @@ export class GenAiConnector extends SubActionConnector<GenAiConfig, GenAiSecrets
   }
 
   public async runApi({ body }: GenAiRunActionParams): Promise<GenAiRunActionResponse> {
+    const sanitizedBody = sanitizeRequest(this.provider, this.url, body);
+    const axiosOptions = getAxiosOptions(this.provider, this.key, false);
     const response = await this.request({
       url: this.url,
       method: 'post',
       responseSchema: GenAiRunActionResponseSchema,
-      data: body,
-      headers: {
-        ...(this.provider === OpenAiProviderType.OpenAi
-          ? { Authorization: `Bearer ${this.key}` }
-          : { ['api-key']: this.key }),
-        ['content-type']: 'application/json',
-      },
+      data: sanitizedBody,
+      ...axiosOptions,
     });
     return response.data;
   }
 
+  public async executeApi({
```

Comment:
What do you think about naming this method more descriptively? `streamApi`? execute seems to be a synonym for run, and it is not immediately apparent what the difference is

Reply:
How about just `testApi` since the subaction is `test`? `streamApi` seems deceptive since you can request a streaming response or not

Reply:
Hm... should the subaction be test? The test refers to this screen:
<img width="1862" alt="Screenshot 2023-07-14 at 11 55 39 AM" src="https://github.com/elastic/kibana/assets/6935300/3fcdb477-b724-4fd0-be4f-d60629f5b893">

Reply:
If it would be less confusing, I can add a new subaction for this stream functionality and leave the test subaction the same as before

Reply:
Updated in https://github.com/elastic/kibana/pull/161676/commits/09b0a3261ca31196f0d4b478e89d8b7fe71fff06

Reply:
appreciate that, thank you!